CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -2364,7 +2368,7 @@ protected function processFormSubmission($contactID) {
         // @todo move premium processing to complete transaction if it truly is an 'after' action.
         $this->postProcessPremium($premiumParams, $result['contribution']);
       }
-      if (!empty($result['contribution'])) {
+      if (($result['payment_status_id'] ?? NULL) == 1) {
```

Comment:
$result['contribution'] is not even used in `completeTransaction()`. Currently it's needed for the contribution ID but it will always be set anyway.